custom_components/tahoma/coordinator.py

```diff
@@ -50,16 +55,36 @@ async def _async_update_data(self) -> Dict[str, Device]:
             raise UpdateFailed(f"Error communicating with the TaHoma API: {exception}")
         else:
             for event in events:
+                if event.name == "DeviceAvailableEvent":
+                    self.devices[event.deviceurl].available = True
+
                 if event.name == "DeviceStateChangedEvent":
                     for state in event.device_states:
                         device = self.devices[event.deviceurl]
                         if state.name not in device.states:
                             device.states[state.name] = state
                         device.states[state.name].value = self._get_state(state)
+
+                if (
+                    event.name == "ExecutionRegisteredEvent"
+                    and event.exec_id in self.executions
+                ):
+                    self.update_interval = timedelta(seconds=1)
+
+                if (
+                    event.name == "ExecutionStateChangedEvent"
+                    and event.exec_id in self.executions
+                ) and event.new_state == "COMPLETED":
+                    self.executions.pop(event.exec_id, None)
+
+            if len(self.executions) < 1:
+                self.update_interval = self.original_update_interval
+
```

Comment:
What happen if I have both ExecutionRegisteredEvent and ExecutionStateChangedEvent at the same time? Did you check if it can occur with 30s poll interval for instance?

Reply:
We should perhaps also add a security in case we miss a COMPLETED event, in which case we will have 1s as poll interval forever.

Reply:
> What happen if I have both ExecutionRegisteredEvent and ExecutionStateChangedEvent at the same time? Did you check if it can occur with 30s poll interval for instance?

Does it matter? If there are no executions left, the update interval will be set to the default interval in the end. I wrote this with a high interval in mind.